lib/assertions.ts

```diff
@@ -430,6 +430,40 @@ export default class DOMAssertions {
     this.doesNotHaveClass(expected, message);
   }
 
+  /**
+   * Assert that the [HTMLElement][] has the `expected` style declarations using
+   * [`window.getComputedStyle`](https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle).
+   *
+   * @name hasStyle
+   * @param {object} expected
+   * @param {string?} message
+   *
+   * @example
+   * assert.dom('.progress-bar').hasStyle({
+   *   opacity: 1,
+   *   display: 'block'
+   * });
+   *
+   * @see {@link #hasClass}
+   */
+  hasStyle(expected: object, message?: string): void {
+    let element = this.findTargetElement();
+    if (!element) return;
+    let actual = {};
+    let result = false;
+    let cssStyleDeclaration = window.getComputedStyle(element);
+    for (let property in expected) {
+      actual[property] = cssStyleDeclaration[property];
+      if (!result) {
+        result = '' + expected[property] === actual[property];
```

Comment:
what is this doing?

maybe something like this might be easier to read/understand:
```js
let computedStyle = window.getComputedStyle(element);

let expectedProperties = Object.keys(expected);
let result = expectedProperties.every(property => computedStyle[property] === expected[property]);

let actual = {};
expectedProperties.forEach(property => actual[property] = computedStyle[property]);
```

Reply:
refactored as suggested 👍